wordpress-mobile/WordPress-Android
# [Compile Warnings As Errors] WordPress Module - Resolve Deprecated Warnings - Part.1

Parent: #17173
Partially Closes: #17181

This PR resolves one part of all `Deprecated` related warnings for the `WordPress` module, and in this part, the most generic ones and specific to the main source:

Main Source (About `247`)
- About `247` warnings x `'XYX' is deprecated. Deprecated in Java`

-----

Warnings Resolution List:

1. [Resolve from html deprecated warnings.](https://github.com/wordpress-mobile/WordPress-Android/pull/17246/commits/542ace1d56d3a6accb29c7fdf2ebe7c16cab46d3)
2. [Resolve on navigation (re) selected deprecated warnings.](https://github.com/wordpress-mobile/WordPress-Android/pull/17246/commits/11e6f278ab4326e627287cd0b65d32e95e2d1413)
3. [Resolve handler deprecated warnings.](https://github.com/wordpress-mobile/WordPress-Android/pull/17246/commits/7ee8cb30fdac64c7e098a6972bda3fe4e3932178)
4. [Resolve lifecycle registry mark state deprecated warning.](https://github.com/wordpress-mobile/WordPress-Android/pull/17246/commits/a3be18f24ecf743800b54d106405ba62807fcfd9)
5. [Resolve nested web view get action masked deprecated warning.](https://github.com/wordpress-mobile/WordPress-Android/pull/17246/commits/2f57fdc0c7e2acc2794fbb5e3d93bba5f15122ae)

Warnings Suppression List:

1. [Suppress on/start activity (for) result deprecated warnings.](https://github.com/wordpress-mobile/WordPress-Android/pull/17246/commits/103b90ccd3664c3295c772d2fcb4b119fc3750cd) (Associated Issue: #17249)
2. [Suppress set/get target fragment deprecated warnings.](https://github.com/wordpress-mobile/WordPress-Android/pull/17246/commits/7ddf2f631312ebf1043fa4dbba1c101f91c0d0b6) (Associated Issue: #17250)
6. [Suppress on activity created deprecated warnings.](https://github.com/wordpress-mobile/WordPress-Android/pull/17246/commits/6e32c33ccf1257205eff2df33f03838439438567) (Associated Issue: #17251)
7. [Suppress fragment pager adapter deprecated warnings.](https://github.com/wordpress-mobile/WordPress-Android/pull/17246/commits/623cf29516ed585831263f26c9493731121fac59) (Associated Issue: #17257)
8. [Suppress progress dialog deprecated warnings.](https://github.com/wordpress-mobile/WordPress-Android/pull/17246/commits/a1de9eb9aa55de319bc909432415e291ffcaa38d) (Associated Issue: #17258)
9. [Suppress comments detail activity deprecated warnings.](https://github.com/wordpress-mobile/WordPress-Android/pull/17246/commits/28bcef782d4541de127d0cfb1662669cc07cfb6e)
10. [Suppress exo player utils deprecated warnings.](https://github.com/wordpress-mobile/WordPress-Android/pull/17246/commits/567e63e0ad3d4487859eb0b9b339ce528384a70e) (Associated Issue: #17259)
11. [Suppress request permissions deprecated warnings.](https://github.com/wordpress-mobile/WordPress-Android/pull/17246/commits/4a037bef101a0e61d2bf69387285a0455a430aa6) (Associated Issue: #17249)
12. [Suppress get external storage public dir deprecated warnings.](https://github.com/wordpress-mobile/WordPress-Android/pull/17246/commits/f1e9bcd4ea0053bef7b21828c2d90df48d77d2f4) (Associated Issue: #17252)
13. [Suppress photo picker activity deprecated warnings.](https://github.com/wordpress-mobile/WordPress-Android/pull/17246/commits/7159db448c0ab5985353cffbb63cd0adbeac4215)
14. [Suppress async task deprecated warnings.](https://github.com/wordpress-mobile/WordPress-Android/pull/17246/commits/954bfaed48bf535db4825f90f9e73ff77ebac26f) (Associated Issue: #17260)
15. [Suppress glide deprecated warnings.](https://github.com/wordpress-mobile/WordPress-Android/pull/17246/commits/538e46c0a9586b9305f0eb920f473f41d317c4b2) (Associated Issue: #17261)
16. [Suppress allow scanning by media scanner deprecated warning.](https://github.com/wordpress-mobile/WordPress-Android/pull/17246/commits/8c76d9e19118d6288256aab7f06598fa8dbe044a) (Associated Issue: #17252)
17. [Suppress window manager display/insets deprecated warnings.](https://github.com/wordpress-mobile/WordPress-Android/pull/17246/commits/a5b8e332aaa79f2bc99ad48e7d57abffab35e29e) (Associated Issue: #17256)
18. [Suppress fragment manager deprecated warnings.](https://github.com/wordpress-mobile/WordPress-Android/pull/17246/commits/75db65bced75dfbd53b0c34797eafa4e4024f6b9) (Associated Issue: #17253)
19. [Suppress set bound in parent deprecated warnings.](https://github.com/wordpress-mobile/WordPress-Android/pull/17246/commits/bff920f3e565a2860df1c95615d434c5132c5999) (Associated Issue: #17254)
20. [Suppress signature deprecated warnings.](https://github.com/wordpress-mobile/WordPress-Android/pull/17246/commits/3751b8fd5b676d97ddab3c02cd1e73d7cf265d2c) (Associated Issue: #17255)

Refactor List:

1. [Extract on result action functions for jetpack remote install.](https://github.com/wordpress-mobile/WordPress-Android/pull/17246/commits/5970d09df0eb754399dd0a6b20942f8135c4ba66)
2. [Replace java with kotlin suppression for deprecation.](https://github.com/wordpress-mobile/WordPress-Android/pull/17246/commits/897977a2dac84ffed576f789c58d02a4c95176fd)

In addition to the above warning resolution list, the below `Detekt` warnings got resolved as well, leaving Detekt's `baseline.xml` file 5 entries shorter:

1. [Resolve detekt warnings by manually updating the baseline.](https://github.com/wordpress-mobile/WordPress-Android/pull/17246/commits/7811dd138074056d0363c5644752d2b883cb6678)

-----

PS: @ovitrif  I added you as the main reviewer, that is, in addition to @wordpress-mobile/apps-infrastructure team itself, but randomly, since I want someone from the WordPress Android team to primarily sign-off on that change. 🥇

FYI: I am going to randomly add more of you in those PRs that will follow, just so you become more aware of this change and how close we are on enabling allWarningsAsErrors by default everywhere. 🎉

-----

To test:

- There is nothing much to test here.
- Verifying that all the CI checks are successful should be enough.
- However, if you want to be thorough about reviewing this change, you could:
    - Quickly smoke test both, the `WordPress` and `Jetpack` apps, and see if they are both working as expected.
    - In addition to the above smoke testing, per the below class refactor, you could do some more targeted testing:
        - [WPMainNavigationView](https://github.com/wordpress-mobile/WordPress-Android/pull/17246/commits/11e6f278ab4326e627287cd0b65d32e95e2d1413):
            1. Make sure the `MySite`, `Reader` and `Notification` tab labels are shown on the selected navigation item.
            2. Add debugger and test the `MySite`, `Reader` and `Notification` tabs to make sure that when selecting a new tab, the `onNavigationItemSelected(...)` function is called, while when re-selecting the existing tab, the `onNavigationItemReselected(...)` function is called.
        - [PostListViewModel](https://github.com/wordpress-mobile/WordPress-Android/pull/17246/commits/a3be18f24ecf743800b54d106405ba62807fcfd9):
            1. Make sure the `Posts` screen works as expected.
        - [NestedWebView](https://github.com/wordpress-mobile/WordPress-Android/pull/17246/commits/2f57fdc0c7e2acc2794fbb5e3d93bba5f15122ae):
            1. Make sure the nested web view related screens and functionality is working as expected.

-----

## Regression Notes

1. Potential unintended areas of impact

See the `targeted testing` part above, which is, within the `To test` section itself.

2. What I did to test those areas of impact (or what existing automated tests I relied on)

See `To test` section above.

3. What automated tests I added (or what prevented me from doing so)

N/A

-----

PR submission checklist:
